admin/services/class-indexable-post-provider.php

```diff
@@ -10,51 +10,88 @@
-			'breadcrumb_title'            => $this->get_meta_value( 'bctitle', $object_id ),
-			'og_title'                    => $this->get_meta_value( 'opengraph-title', $object_id ),
-			'og_description'              => $this->get_meta_value( 'opengraph-description', $object_id ),
-			'og_image'                    => $this->get_meta_value( 'opengraph-image', $object_id ),
-			'twitter_title'               => $this->get_meta_value( 'twitter-title', $object_id ),
-			'twitter_description'         => $this->get_meta_value( 'twitter-description', $object_id ),
-			'twitter_image'               => $this->get_meta_value( 'twitter-image', $object_id ),
-			'is_robots_noindex'           => $this->get_robots_noindex_value( $this->get_meta_value( 'meta-robots-noindex', $object_id ) ),
-			'is_robots_nofollow'          => $this->get_meta_value( 'meta-robots-nofollow', $object_id ) === '1',
-			'is_robots_noarchive'         => strpos( $this->get_meta_value( 'meta-robots-adv', $object_id ), 'noarchive' ) !== false,
-			'is_robots_noimageindex'      => strpos( $this->get_meta_value( 'meta-robots-adv', $object_id ), 'noimageindex' ) !== false,
-			'is_robots_nosnippet'         => strpos( $this->get_meta_value( 'meta-robots-adv', $object_id ), 'nosnippet' ) !== false,
-			'primary_focus_keyword'       => $this->get_meta_value( 'focuskw', $object_id ),
-			'primary_focus_keyword_score' => (int) $this->get_meta_value( 'linkdex', $object_id ),
-			'readability_score'           => (int) $this->get_meta_value( 'content_score', $object_id ),
-			'is_cornerstone'              => $this->get_meta_value( 'is_cornerstone', $object_id ) === '1',
-			'link_count'                  => (int) $link_count->get( $object_id ),
-			'incoming_link_count'         => (int) $link_count->get( $object_id, 'incoming_link_count' ),
-			'created_at'                  => null,
-			'updated_at'                  => null,
-		);
+		$indexable = WPSEO_Post_Indexable::from_object( $object_id );
+
+		if ( $as_object === true ) {
+			return $indexable;
+		}
+
+		return $indexable->to_array();
+	}
+
+	/**
+	 * Handles the patching of values for an existing indexable.
+	 *
+	 * @param int 	$object_id 		The ID of the object.
+	 * @param array $requestdata 	The request data to store.
+	 *
+	 * @return void
+	 *
+	 * @throws WPSEO_Invalid_Indexable_Exception The invalid argument exception.
+	 * @throws Exception 						 Exception that is thrown if patching the object has failed.
+	 */
+	public function patch( $object_id, $requestdata ) {
+		$indexable = $this->get( $object_id, true );
+
+		if ( $indexable === array() ) {
+			throw WPSEO_Invalid_Indexable_Exception::non_existing_indexable( $object_id );
+		}
+
+		$new_indexable    = $indexable->update( $requestdata );
+		$stored_indexable = $this->store_indexable( $new_indexable );
+
+		if ( $stored_indexable === true ) {
+			return;
+		}
+
+		throw new Exception( 'Patch failed' );
```

Comment:
This should preferably return a more accurate message, more importantly it should be translatable.